subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskExecutionIntegrationTest.groovy

```diff
@@ -691,4 +692,190 @@ task someTask(dependsOn: [someDep, someOtherDep])
         expect:
         succeeds "custom", "--rerun-tasks"
     }
+
+    def "only must run after"() {
```

Comment:
Those tests seem to be passing on master, should we include them anyway or we can drop them?

Reply:
I think this test can go. The one about the software model should stay. It passes on `master` since my changes have been reverted there.

Reply:
See #8715 and gradle/gradle@18b6c927b38e48eed57a5db28f9939ceb84de3d4. When #8715 is re-introduced, then the software model test will fail again. And we want to re-introduce #8715 when you are done with your changes.

Reply:
I create this PR https://github.com/gradle/gradle/pull/10050 Is that enough?

Reply:
I think that is enough.

Reply:
The other PR prevent the creation of duplicate task at creation. All other tests should be included when re-introducing #8715.